src/clock.rs

```diff
@@ -0,0 +1,119 @@
+use std::sync::atomic::AtomicU64;
+use std::time::{Duration, Instant as StdInstant};
+
+use rustc_data_structures::sync::Ordering;
```

Comment:
That's a funny place to import this from. We should import this from `std::sync::atomic`. Does rustc suggest this import?

Reply:
indeed :p